src/wp-includes/deprecated.php

```diff
@@ -6422,3 +6422,150 @@ function wp_create_block_style_variation_instance_name( $block, $variation ) {
+ * @param int    $new_term_id      ID of the new term created for the $term_taxonomy_id.
+ * @param int    $term_taxonomy_id ID for the term_taxonomy row affected by the split.
+ * @param string $taxonomy         Taxonomy for the split term.
+ */
+function _wp_check_split_default_terms( $term_id, $new_term_id, $term_taxonomy_id, $taxonomy ) {
+}
+
+/**
+ * Checks menu items when a term gets split to see if any of them need to be updated.
+ *
+ * This function is non-operational since x.y.z.
+ *
+ * @ignore
+ * @since 4.2.0
+ * @deprecated x.y.z Shared term functionality no longer exists.
+ *
+ * @param int    $term_id          ID of the formerly shared term.
+ * @param int    $new_term_id      ID of the new term created for the $term_taxonomy_id.
+ * @param int    $term_taxonomy_id ID for the term_taxonomy row affected by the split.
+ * @param string $taxonomy         Taxonomy for the split term.
+ */
+function _wp_check_split_terms_in_menus( $term_id, $new_term_id, $term_taxonomy_id, $taxonomy ) {
+}
+
+/**
+ * If the term being split is a nav_menu, changes associations.
+ *
+ * This function is non-operational since x.y.z.
+ *
+ * @ignore
+ * @since 4.3.0
+ * @deprecated x.y.z Shared term functionality no longer exists.
+ *
+ * @param int    $term_id          ID of the formerly shared term.
+ * @param int    $new_term_id      ID of the new term created for the $term_taxonomy_id.
+ * @param int    $term_taxonomy_id ID for the term_taxonomy row affected by the split.
+ * @param string $taxonomy         Taxonomy for the split term.
+ */
+function _wp_check_split_nav_menu_terms( $term_id, $new_term_id, $term_taxonomy_id, $taxonomy ) {
+}
+
+/**
+ * Gets data about terms that previously shared a single term_id, but have since been split.
+ *
+ * This function is non-operational since x.y.z.
+ *
+ * @since 4.2.0
+ * @deprecated x.y.z Shared term functionality no longer exists.
+ *
+ * @param int $old_term_id Term ID. This is the old, pre-split term ID.
+ * @return array Array of new term IDs, keyed by taxonomy.
+ */
+function wp_get_split_terms( $old_term_id ) {
+	return array();
+}
+
+/**
+ * Gets the new term ID corresponding to a previously split term.
+ *
+ * This function is non-operational since x.y.z.
+ *
+ * @since 4.2.0
+ * @deprecated x.y.z Shared term functionality no longer exists.
+ *
+ * @param int    $old_term_id Term ID. This is the old, pre-split term ID.
+ * @param string $taxonomy    Taxonomy that the term belongs to.
+ * @return int Returns the term ID.
+ */
+function wp_get_split_term( $old_term_id, $taxonomy ) {
+	return $old_term_id;
+}
+
+/**
+ * Determines whether a term is shared between multiple taxonomies.
+ *
+ * This function is non-operational since x.y.z.
+ *
+ * @since 4.4.0
+ * @deprecated x.y.z Shared term functionality no longer exists.
+ *
+ * @param int $term_id Term ID.
+ * @return false Returns false.
+ */
+function wp_term_is_shared( $term_id ) {
+	return false;
+}
```

Comment:
Shouldn't these trigger deprecation warnings via `_deprecated_function()`? I think at least for the public (not-underscore-prefixed) functions we should make that more obvious via warnings.

Reply:
Yep good spot 👍